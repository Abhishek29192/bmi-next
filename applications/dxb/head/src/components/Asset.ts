import { graphql } from "gatsby";

export type Data = {
  resize: {
    src: string;
  };
  file: {
    fileName: string;
    url: string;
  };
};

export const query = graphql`
  fragment AssetFragment on ContentfulAsset {
    fluid {
      ...GatsbyContentfulFluid
    }
    resize(width: 1000, toFormat: JPG, jpegProgressive: true, quality: 60) {
      src
    }
    file {
      fileName
      url
    }
  }
`;
