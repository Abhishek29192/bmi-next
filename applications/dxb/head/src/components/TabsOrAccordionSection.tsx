import {
  Accordion,
  AccordionSummaryProps,
  replaceSpaces,
  Section,
  Tabs,
  transformHyphens,
  Typography
} from "@bmi-digital/components";
import Tab, { TabProps } from "@mui/material/Tab";
import { graphql } from "gatsby";
import React from "react";
import RichText from "../components/RichText";
import withGTM from "../utils/google-tag-manager";
import { Data as TitleWithContentData } from "./TitleWithContent";

const GTMAccordionSummary = withGTM<AccordionSummaryProps>(Accordion.Summary);

export type Data = {
  __typename: "ContentfulTabsOrAccordionSection";
  title: string | null;
  description: null | { description: string };
  items: readonly TitleWithContentData[];
  type: "Accordion" | "Tabs";
};

const SectionAccordion = ({
  items
}: {
  items: readonly TitleWithContentData[];
}) => {
  return (
    <>
      {items.length > 0 && (
        <Accordion>
          {items.map(({ title, name, content }) => {
            const newTitle = title || name;
            return (
              <Accordion.Item key={title}>
                <GTMAccordionSummary
                  gtm={{
                    id: "selector-accordion1",
                    label: newTitle,
                    action: "Selector – Accordion"
                  }}
                >
                  <Typography component="h3" variant="h6">
                    {transformHyphens(newTitle)}
                  </Typography>
                </GTMAccordionSummary>
                <Accordion.Details>
                  <RichText document={content} hasNoBottomMargin />
                </Accordion.Details>
              </Accordion.Item>
            );
          })}
        </Accordion>
      )}
    </>
  );
};

const SectionTabs = ({ items }: { items: readonly TitleWithContentData[] }) => {
  const GTMTab = withGTM<TabProps>(Tab, {
    label: "label"
  });

  return (
    <Tabs
      tabComponent={(props: TabProps) => (
        <GTMTab
          gtm={{ id: "selector-tabs2", action: "Selector – Tabs" }}
          {...props}
        />
      )}
      color="primary"
      initialValue={items.length > 0 ? items[0].title || items[0].name : null}
    >
      {items.length > 0 &&
        items.map(({ title, content, name }) => {
          const newTitle = title || name;
          return (
            <Tabs.TabPanel heading={newTitle} index={newTitle} key={newTitle}>
              <RichText document={content} />
            </Tabs.TabPanel>
          );
        })}
    </Tabs>
  );
};

const componentMap = {
  Accordion: SectionAccordion,
  Tabs: SectionTabs
};

const TabsOrAccordionSection = ({ data }: { data: Data }) => {
  const { title, type, items, description } = data;
  const backgroundColor = type === "Tabs" ? "pearl" : "white";
  // eslint-disable-next-line security/detect-object-injection
  const Component = componentMap[type];

  return (
    <Section
      backgroundColor={backgroundColor}
      data-testid={`tabs-or-accordion-section-${replaceSpaces(title)}`}
    >
      {title && <Section.Title>{title}</Section.Title>}
      {description && (
        <Typography variant="body1">{description.description}</Typography>
      )}
      <div style={{ marginTop: "40px" }}>
        <Component items={items} />
      </div>
    </Section>
  );
};

export default TabsOrAccordionSection;

export const query = graphql`
  fragment TabsOrAccordionSectionFragment on ContentfulTabsOrAccordionSection {
    __typename
    type
    title
    description {
      description
    }
    items {
      ...TitleWithContentFragment
    }
  }
`;
