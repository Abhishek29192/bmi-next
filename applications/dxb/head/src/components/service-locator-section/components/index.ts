// istanbul ignore file: doesn't hold any logic
export * from "./CompanyDetails";
export * from "./IntegratedLinkCard";
export * from "./ServiceLocationBlock/SearchLocationBlock";
export * from "./ServiceLocatorChips/ServiceLocatorChips";
export * from "./ServiceLocatorMap";
export * from "./ServiceLocatorResultList/ServiceLocatorResultList";
export * from "./useScrollTo";
