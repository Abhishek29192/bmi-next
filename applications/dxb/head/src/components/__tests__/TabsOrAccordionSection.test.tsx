import { ThemeProvider } from "@bmi-digital/components";
import { render, screen } from "@testing-library/react";
import React from "react";
import { BLOCKS } from "@contentful/rich-text-types";
import TabsOrAccordionSection, { Data } from "../TabsOrAccordionSection";

const contentMock = JSON.stringify({
  nodeType: BLOCKS.DOCUMENT,
  data: {},
  content: [
    {
      nodeType: BLOCKS.PARAGRAPH,
      content: [
        {
          nodeType: "text",
          value:
            "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s",
          marks: [],
          data: {}
        }
      ],
      data: {}
    }
  ]
});

const getDummyData = ({ name, title, type }): Data => {
  return {
    __typename: "ContentfulTabsOrAccordionSection",
    description: { description: "string" },
    items: [
      {
        __typename: "ContentfulTitleWithContent",
        name: name,
        title: title,
        content: {
          raw: contentMock,
          references: []
        }
      }
    ],
    title: "string",
    type: type
  };
};
const dataWithNoItems: Data = {
  __typename: "ContentfulTabsOrAccordionSection",
  description: { description: "string" },
  items: [],
  title: "string",
  type: "Tabs"
};

describe("TabsOrAccordionSection", () => {
  it("Accordion renders Correctly ", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection
          data={getDummyData({
            name: "name",
            title: "hello",
            type: "Accordion"
          })}
        />
      </ThemeProvider>
    );
    expect(screen.getByText("hello")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("Tabs renders Correctly ", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection
          data={getDummyData({ name: "name", title: "hello", type: "Tabs" })}
        />
      </ThemeProvider>
    );
    expect(screen.getByText("hello")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("Tabs default to name if title is null", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection
          data={getDummyData({ name: "name", title: null, type: "Tabs" })}
        />
      </ThemeProvider>
    );
    expect(screen.getByText("name")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("Accordion default to name if title is null", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection
          data={getDummyData({ name: "name", title: null, type: "Accordion" })}
        />
      </ThemeProvider>
    );
    expect(screen.getByText("name")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("title attribute is rendered if both name and title exists for tab panels", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection
          data={getDummyData({ name: "name", title: "title", type: "Tabs" })}
        />
      </ThemeProvider>
    );
    expect(screen.queryByText("name")).not.toBeInTheDocument();
    expect(screen.getByText("title")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("not render tab panel if item array is empty", () => {
    const { container } = render(
      <ThemeProvider>
        <TabsOrAccordionSection data={dataWithNoItems} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
