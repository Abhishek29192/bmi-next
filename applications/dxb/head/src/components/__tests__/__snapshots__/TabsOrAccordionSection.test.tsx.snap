// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsOrAccordionSection Accordion default to name if title is null 1`] = `
<div>
  <div
    class="Section-root Section-white css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-1u48ai"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1g8jyxy-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root Accordion-summaryRoot MuiAccordionSummary-gutters css-1b94k5e-MuiButtonBase-root-MuiAccordionSummary-root"
                data-gtm="{\\"id\\":\\"selector-accordion1\\",\\"label\\":\\"name\\",\\"action\\":\\"Selector – Accordion\\"}"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content Accordion-summaryContent MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                >
                  <h3
                    class="MuiTypography-root MuiTypography-h6  css-1le86ss-MuiTypography-root"
                    data-testid="typography"
                  >
                    name
                  </h3>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root  css-glpxpj-MuiAccordionDetails-root"
                      >
                        <div
                          class="RichText-root css-17t4j5c"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom  css-abi5v8-MuiTypography-root"
                            data-testid="typography"
                          >
                            Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabsOrAccordionSection Accordion renders Correctly  1`] = `
<div>
  <div
    class="Section-root Section-white css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-1u48ai"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1g8jyxy-MuiPaper-root-MuiAccordion-root"
            >
              <div
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root Accordion-summaryRoot MuiAccordionSummary-gutters css-1b94k5e-MuiButtonBase-root-MuiAccordionSummary-root"
                data-gtm="{\\"id\\":\\"selector-accordion1\\",\\"label\\":\\"hello\\",\\"action\\":\\"Selector – Accordion\\"}"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content Accordion-summaryContent MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                >
                  <h3
                    class="MuiTypography-root MuiTypography-h6  css-1le86ss-MuiTypography-root"
                    data-testid="typography"
                  >
                    hello
                  </h3>
                </div>
                <div
                  class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiAccordion-region"
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root  css-glpxpj-MuiAccordionDetails-root"
                      >
                        <div
                          class="RichText-root css-17t4j5c"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom  css-abi5v8-MuiTypography-root"
                            data-testid="typography"
                          >
                            Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabsOrAccordionSection Tabs default to name if title is null 1`] = `
<div>
  <div
    class="Section-root Section-pearl css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-fpjlvn"
            data-testid="tabs"
          >
            <div
              class="Tabs-tabsBar"
            >
              <div
                class=" css-ze8cyt"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXl css-1ozv57h-MuiContainer-root"
                >
                  <div
                    class="MuiTabs-root css-7cgbzd-MuiTabs-root"
                    color="primary"
                  >
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="tabs"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="tabpanel-name"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected Tabs-tabRoot css-15jz8p0-MuiButtonBase-root-MuiTab-root"
                          data-gtm="{\\"id\\":\\"selector-tabs2\\",\\"label\\":\\"name\\",\\"action\\":\\"Selector – Tabs\\"}"
                          data-testid="tab-name"
                          id="tab-name"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          name
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="MuiTabs-indicator css-tv859c-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tab-name"
              class="Tabs-tabPanel Tabs-tabPanelBox MuiBox-root css-19kzrtu"
              data-testid="tabpanel-name"
              heading="name"
              id="tabpanel-name"
              role="tabpanel"
            >
              <div
                class="RichText-root css-17t4j5c"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom  css-abi5v8-MuiTypography-root"
                  data-testid="typography"
                >
                  Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabsOrAccordionSection Tabs renders Correctly  1`] = `
<div>
  <div
    class="Section-root Section-pearl css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-fpjlvn"
            data-testid="tabs"
          >
            <div
              class="Tabs-tabsBar"
            >
              <div
                class=" css-ze8cyt"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXl css-1ozv57h-MuiContainer-root"
                >
                  <div
                    class="MuiTabs-root css-7cgbzd-MuiTabs-root"
                    color="primary"
                  >
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="tabs"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="tabpanel-hello"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected Tabs-tabRoot css-15jz8p0-MuiButtonBase-root-MuiTab-root"
                          data-gtm="{\\"id\\":\\"selector-tabs2\\",\\"label\\":\\"hello\\",\\"action\\":\\"Selector – Tabs\\"}"
                          data-testid="tab-hello"
                          id="tab-hello"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          hello
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="MuiTabs-indicator css-tv859c-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tab-hello"
              class="Tabs-tabPanel Tabs-tabPanelBox MuiBox-root css-19kzrtu"
              data-testid="tabpanel-hello"
              heading="hello"
              id="tabpanel-hello"
              role="tabpanel"
            >
              <div
                class="RichText-root css-17t4j5c"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom  css-abi5v8-MuiTypography-root"
                  data-testid="typography"
                >
                  Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabsOrAccordionSection not render tab panel if item array is empty 1`] = `
<div>
  <div
    class="Section-root Section-pearl css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-fpjlvn"
            data-testid="tabs"
          >
            <div
              class="Tabs-tabsBar"
            >
              <div
                class=" css-ze8cyt"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXl css-1ozv57h-MuiContainer-root"
                >
                  <div
                    class="MuiTabs-root css-7cgbzd-MuiTabs-root"
                    color="primary"
                  >
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="tabs"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      />
                      <span
                        class="MuiTabs-indicator css-tv859c-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabsOrAccordionSection title attribute is rendered if both name and title exists for tab panels 1`] = `
<div>
  <div
    class="Section-root Section-pearl css-cxq310"
  >
    <div
      class=" css-ze8cyt"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-15p7r21-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 Typography-underline css-je89nd-MuiTypography-root"
          data-testid="typography"
        >
          string
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1  css-87ejlm-MuiTypography-root"
          data-testid="typography"
        >
          string
        </p>
        <div
          style="margin-top: 40px;"
        >
          <div
            class=" css-fpjlvn"
            data-testid="tabs"
          >
            <div
              class="Tabs-tabsBar"
            >
              <div
                class=" css-ze8cyt"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXl css-1ozv57h-MuiContainer-root"
                >
                  <div
                    class="MuiTabs-root css-7cgbzd-MuiTabs-root"
                    color="primary"
                  >
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="tabs"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="tabpanel-title"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected Tabs-tabRoot css-15jz8p0-MuiButtonBase-root-MuiTab-root"
                          data-gtm="{\\"id\\":\\"selector-tabs2\\",\\"label\\":\\"title\\",\\"action\\":\\"Selector – Tabs\\"}"
                          data-testid="tab-title"
                          id="tab-title"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          title
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="MuiTabs-indicator css-tv859c-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                    <div
                      class="MuiButtonBase-root MuiTabScrollButton-root MuiTabScrollButton-horizontal Mui-disabled MuiTabs-scrollButtons MuiTabs-scrollButtonsHideMobile tabsScrollBtn css-145v6pe-MuiButtonBase-root-MuiTabScrollButton-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tab-title"
              class="Tabs-tabPanel Tabs-tabPanelBox MuiBox-root css-19kzrtu"
              data-testid="tabpanel-title"
              heading="title"
              id="tabpanel-title"
              role="tabpanel"
            >
              <div
                class="RichText-root css-17t4j5c"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom  css-abi5v8-MuiTypography-root"
                  data-testid="typography"
                >
                  Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
